### PR TITLE
Cooler zombie moths

### PR DESCRIPTION
**Pull request by Simyon264**
## About the PR

Zombie moths now say "Clothessss", "Shooooesss" or something similar instead of "Brainsss"
## Why / Balance

See this [discord forum post](https://canary.discord.com/channels/310555209753690112/1175040570939408404). Also it's cool.
## Technical details

Adds a new accent called "zombieMoth". Added "ZombieAccentOverrideComponent which is used in the accent assignment in "ZombieSystem.Transform.cs".
## Media

https://github.com/space-wizards/space-station-14/assets/63975668/a046b40a-a16c-49c5-9772-207454c01b8c
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: Simyon
- tweak: Moths that are turned into zombies will now prefer cloth instead of brains.

### CODE REVIEW

File: Content.Server/Zombies/ZombieSystem.Transform.cs [hunk 1]
```diff
@@ -101,7 +101,14 @@ public void ZombifyEntity(EntityUid target, MobStateComponent? mobState = null)
             RemComp<ThirstComponent>(target);
 
             //funny voice
-            EnsureComp<ReplacementAccentComponent>(target).Accent = "zombie";
+            var accentType = Prototype(target)!.ID switch
+            {
+                "MobMoth" => "zombieMoth",
+                "MobMothroach" => "zombieMoth",
+                _ => "zombie"
+            };
```
Comment (moonheart08) [inline Content.Server/Zombies/ZombieSystem.Transform.cs:109]: Absolutely never use prototype checks, they're disallowed. Add a component to moths instead.

**Review by moonheart08 — CHANGES_REQUESTED** [review]

**Comment by BYONDFuckery** [comment]
does moth accent changes in 4 commits

me: cargo crate in 14 commits

**Comment by BYONDFuckery** [comment]
would be neat if it got merged though

File: Content.Server/Zombies/ZombieSystem.Transform.cs (hunk 1 above)
Comment (Simyon264) [reply Content.Server/Zombies/ZombieSystem.Transform.cs]: Changed it to a component.

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**